src/arguments.toit

```diff
@@ -228,26 +228,23 @@ class Arguments:
 // ----------------------------------------------------------------------------
 
 // Argument parsing functionality.
-parse_ grammar command arguments index:
-  if not command and index < arguments.size:
-    first := arguments[index]
-    grammar.commands_.get first --if_present=:
-      sub := it
-      return parse_ sub first arguments index + 1
-
+parse_ grammar command arguments index --options={:}:
   // Populate the options from the default values or empty lists (for multi-options)
-  options := {:}
   rest := []
-  grammar.options_.do --values:
-    if it.is_multi_option:
-      options[it.name] = []
-    else:
-      options[it.name] = it.default
+  grammar.options_.do --values: | option |
+    options.get option.name --init=:
+      option.is_multi_option ? [] : option.default
 
   seen_options := {}
 
   while index < arguments.size:
     argument := arguments[index]
+
+    if not command and index < arguments.size:
+      grammar.commands_.get argument --if_present=:
```

Comment:
Maybe using `sub := grammar.commands_.get argument` and do `if sub: ...` in the next line?